perf/src/sigverify.rs

```diff
@@ -420,34 +423,76 @@ pub fn generate_offsets(
     )
 }
 
-fn dedup_packet(count: &AtomicU64, packet: &mut Packet, bloom: &AtomicBloom<&[u8]>) {
-    // If this packet was already marked as discard, drop it
-    if packet.meta.discard() {
-        return;
+pub struct Deduper {
+    filter: Vec<AtomicU64>,
+    seed: (u128, u128),
+    age: Instant,
+    max_age: Duration,
+    pub saturated: AtomicBool,
+}
+
+impl Deduper {
+    pub fn new(size: u32, max_age_ms: u64) -> Self {
```

Comment:
To make this more explicit, `max_age` could be changed to be of type `Duration`. Then there's no risk of a user accidentally passing the wrong u64 here, or someone messing up the conversion to milliseconds.

`MAX_DEDUPER_AGE_MS` can then become:

```rust
const MAX_DEDUPER_AGE: Duration = Duration::from_secs(2);
```